src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import java.util.function.BooleanSupplier;
+import com.chopshop166.chopshoplib.commands.SmartSubsystemBase;
+import com.chopshop166.chopshoplib.motors.Modifier;
+import com.chopshop166.chopshoplib.motors.PIDControlType;
+import com.chopshop166.chopshoplib.motors.SmartMotorController;
+import com.chopshop166.chopshoplib.states.SpinDirection;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.maps.RobotMap.IntakeMap;
+
+public class Intake extends SmartSubsystemBase {
+
+    private final SmartMotorController deploymentMotor;
+    private final SmartMotorController rollerMotor;
+
+    private final BooleanSupplier insideLimit;
+    private final BooleanSupplier outsideLimit;
+
+    private static final double ROLLER_SPEED = 1;
+    private static final double DEPLOY_SPEED = 1;
+
+    private final Modifier insideModifier;
+    private final Modifier outsideModifier;
+
+    public Intake(final IntakeMap map) {
+        this.deploymentMotor = map.getDeploy();
+        this.rollerMotor = map.getRoller();
+        this.insideLimit = map.getInsideLimit();
+        this.outsideLimit = map.getOutsideLimit();
+        this.outsideModifier = Modifier.upperLimit(outsideLimit);
+        this.insideModifier = Modifier.lowerLimit(insideLimit);
+
+        deploymentMotor.setControlType(PIDControlType.SmartMotion);
+        deploymentMotor.setSetpoint(0.25);
```

Comment:
Probably shouldn't set this in the constructor - beware of wild movement

Reply:
where would a good place for it be?

Reply:
Instead of `deploymentMotor.set(xyz)` you'd use `setSetpoint(xyz)` - you're setting a position to move to, not a speed